.github/workflows/rebuilder.py

```diff
@@ -0,0 +1,335 @@
+    >>> d = {"name" : "d", "builds" : { "latest" : { "id" : 2, "state" : "canceled", "submitted_on" : 1731457321 } , "latest_succeeded" : { "id" : 1 } } }
+    >>> e = {"name" : "e", "builds" : { "latest" : { "id" : 2, "state" : "failed", "submitted_on" : 1 } , "latest_succeeded" : { "id" : 1 } } }
+    >>> f = {"name" : "f", "builds" : { "latest" : { "id" : 2, "state" : "failed", "submitted_on" : 1731457321 } , "latest_succeeded" : { "id" : 1 } } }
+    >>> copr_pkgs=[ a, b, c, d, e, f ]
+    >>> project_owner = "@fedora-llvm-team"
+    >>> project_name = "fedora41-clang-20"
+    >>> regressions = get_monthly_rebuild_regressions(project_owner, project_name, datetime.datetime.fromisoformat("2024-11-11"), copr_pkgs)
+    >>> print(regressions)
+    [{'name': 'f', 'url': 'https://copr.fedorainfracloud.org/coprs/@fedora-llvm-team/fedora41-clang-20/build/2/'}]
+
+    """
+    pkgs = []
+    for p in copr_pkgs:
+        latest_succeeded = p["builds"]["latest_succeeded"]
+        latest = p["builds"]["latest"]
+
+        if not latest:
+            continue
+
+        # Don't report regressions if there are still builds in progress
+        if latest["state"] not in [
+            "succeeded",
+            "forked",
+            "skipped",
+            "failed",
+            "canceled",
+        ]:
+            continue
+
+        if not latest_succeeded:
+            continue
+        if latest["id"] == latest_succeeded["id"]:
+            continue
+        # latest is a successful build, but this doesn't mean it failed.
+        # It could be in progress.
+        if latest["state"] != "failed":
+            continue
+        if int(latest["submitted_on"]) < start_time.timestamp():
+            continue
+        latest["name"] = p["name"]
+        pkgs.append(
+            {
+                "name": p["name"],
+                "url": f"https://copr.fedorainfracloud.org/coprs/{project_owner}/{project_name}/build/{latest['id']}/",
+            }
+        )
+    return pkgs
+
+
+def start_rebuild(
+    project_owner: str,
+    project_name: str,
+    copr_client: copr.v3.Client,
+    pkgs: set[str],
+    snapshot_project_name: str,
+):
+
+    print("START", pkgs, "END")
+    # Update the rebuild project to use the latest snapshot
+    copr_client.project_proxy.edit(
+        project_owner,
+        project_name,
+        additional_repos=[
+            "copr://tstellar/fedora-clang-default-cc",
+            f"copr://@fedora-llvm-team/{snapshot_project_name}",
+        ],
+    )
+
+    buildopts = {
+        "background": True,
+    }
+    logging.info("Rebuilding", len(pkgs), "packages")
+    for p in pkgs:
+        logging.info("Rebuild", p)
+        copr_client.build_proxy.create_from_distgit(
+            project_owner, project_name, p, "f41", buildopts=buildopts
+        )
+
+
+def select_snapshot_project(
+    copr_client: copr.v3.Client, target_chroots: list[str], max_lookback_days: int = 14
+) -> str:
```

Comment:
```suggestion
def select_snapshot_project(
    copr_client: copr.v3.Client, target_chroots: list[str], max_lookback_days: int = 14
) -> str | None:
```

The function may return `None`.